### PR TITLE
RadioGroup: Add Badge prop

### PR DESCRIPTION
**Pull request by eniomoura**
#### What changed?

RadioGroup now has a prop to add a Badge next to the label.

#### Why?

To satisfy the use case where users need both a Badge and a helperText with a RadioGroup.RadioButton option, as described in the component request: https://jira.pinadmin.com/browse/GESTALT-7505

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-7505)

### CODE REVIEW

File: packages/gestalt/src/RadioGroupButton.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // @flow strict
 import { type AbstractComponent, forwardRef, type Node as ReactNode, useState } from 'react';
 import classnames from 'classnames';
+import Badge from './Badge';
```
Comment (AlbertCarreras) [inline packages/gestalt/src/RadioGroupButton.js:4]: Can we add an example in the Docs so we can see how it looks like?

File: docs/pages/web/radiogroup.js [hunk 2]
```diff
@@ -288,6 +289,18 @@ export default function DocsPage({
...
         </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Adding a Badge"
+          description={`
+    The \`badge\` prop can be used to add a badge to the RadioButton.
+  `}
+        >
+          <MainSection.Card
+            sandpackExample={
+              <SandpackExample name="Adding a popover example" code={addingABadgeExample} />
```
Comment (AlbertCarreras) [inline docs/pages/web/radiogroup.js:300]: Can we rename this to just code={badge}.   ??

Comment (AlbertCarreras) [reply docs/pages/web/radiogroup.js:300]: name="With Badge"

File: docs/pages/web/radiogroup.js [hunk 3]
```diff
@@ -288,6 +289,18 @@ export default function DocsPage({
             }
           />
         </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Adding a Badge"
```
Comment (AlbertCarreras) [inline docs/pages/web/radiogroup.js:293]: title="With Badge"

File: packages/gestalt/src/RadioGroupButton.js (hunk 1 above)
Comment (eniomoura) [reply packages/gestalt/src/RadioGroupButton.js:4]: Done

File: docs/pages/web/radiogroup.js (hunk 2 above)
Comment (eniomoura) [reply docs/pages/web/radiogroup.js]: Done

**Review by AlbertCarreras — APPROVED** [review]
LGTM

**Pull request merged**